src/gleam/string.gleam

```diff
@@ -96,39 +96,94 @@ pub external fn compare(String, String) -> order.Order =
   "gleam_stdlib" "compare_strings"
 
 /// ## Get Substrings
-
-// TODO
-/// Take a substring given a start and end Grapheme indexes. Negative indexes
-/// are taken starting from the *end* of the list.
+///
+/// Take a substring given a starting index and length.
+/// A negative start or length will return an empty string.
+/// Works on Unicode Graphemes.
 ///
 /// ## Examples
 /// ```gleam
-/// slice("gleam", from: 1, to: 3) == "lea"
-/// slice("gleam", from: 1, to: 10) == "leam"
-/// slice("snakes on a plane!", from: -6, to: -1) == "plane"
+/// slice("gleam", from: 1, length: 3) == "lea"
+/// slice("gleam", from: 1, length: 10) == "leam"
+/// slice("snakes on a plane!", from: -6, to: -1) == ""
 /// ```
 ///
-// pub fn slice(out_of string: String, from start: Int, end: Int) -> String {}
+external fn erl_string_slice(string, start, length) -> String =
+  "string" "slice"
 
-// TODO
-/// Drop *n* Graphemes from the left side of a
+pub fn slice(out_of out_of: String, from from: Int, length length: Int) -> String {
+  let zero = 0
+
+  case from, length {
+    fr, _ if zero > fr ->
+      ""
+
+    _, ln if zero > ln ->
+      ""
+
+    fr, ln ->
+      erl_string_slice(out_of, fr, ln)
+  }
+
+}
+
+/// Drop *n* Graphemes from the start of a string
+///
+/// If `up_to` is <= 0 the input string will be returned unchanged.
+/// If `up_to` is >= the input strings length, "" will be returned.
+/// Works on Unicode Graphemes.
 ///
 /// ## Examples
 /// ```gleam
 /// drop_left(from: "The Lone Gunmen", up_to: 2) == "e Lone Gunmen"
 /// ```
 ///
-// pub fn drop_left(from string: String, up_to num_graphemes: Int) -> String {}
+external fn erl_string_slice_to_infinity(string: String, start: Int) -> String =
```

Comment:
The doc comment above is attached to the helper function rather than the public function